### ISSUE TITLE
Sprint Aug 22 - Sept 2

### ISSUE THREAD

**Issue by timgl** [PostHog/posthog#11341]
# Global Sprint Planning

## 3 things that might take us down

Last week:
1. Rate limit API requests. We're emailing people now. @rcmarron - deployed something that logs, we'll flip the switch to cancel
2. Session recordings taking us down. Let's cap # recordings/day @rcmarron - No progress.
3. Same problem, but events @kappa90 

This week:
1. Enforcing 2FA on github/gsuite/DO/NPM/Ruby gems. @timgl to do this.
2. Cap # of recordings/day or events/day. @rcmarron to write something up.

## Retro: Status of Outcomes from Previous Sprint

1. s3 backed recordings @rcmarron: Status: Hung up because of "zero copy replication" issues. Seems like a feature in development from ClickHouse, and not a massive feature. I'd love to pause this project to see if ClickHouse can catch up.
2. turbo mode wrap up @neilkakkar: 'rapped up!
3. Session Recordings Experience enhancement — Event list @alexkim205 (was support hero last week). Roll over into next week
4. Person on events rollout @EDsCODE Status: In progress. Somewhat backed up from turbo mode rollout.
5. Switch over events to EKS and set ECS to 0 - @guidoiaquinti. Saw increase in 500s. Roll over, mitigate 500s and try again.
6. "Right to be forgotten" tooling - make deleting person/groups/teams data easy @macobo - Persons/Teams waiting for reviews, groups got delayed a bit due to schema changes required.
7. Implement "fundamental" Design System components @benjackwhite. 12/18 components done.
8. Low hanging fruit for visualisations https://github.com/PostHog/posthog/issues/11106 @Twixes. Shipped number. Pushed demo instead.

## Retro: What can we do better next sprint?

1. Eric: Making sure to tighten up goals. We should have clearer milestones.
2. Neil: We don't do spikes. It's worth doing them.
3. Ben: We noticed there were a lot of bugs. Keep a list/board of priority bugs.
4.
5.
6.

## Support hero this sprint

Week 1: Tim
Week 2: Paul

## Plan: Proposed Goals for Next Sprint 

Each goal should have a single owner. Owner can only be an engineer.

1. Call the persons/groups on events done (from the ingestion side). Documentation/async migration/0.8% of events.  @tiina303  
Why? To speed up queries and scale
2. Unblocking large historical exports, fix the current issues, async server is struggling, benchmarking. @macobo
Why? Enabling a large customer
3. We are at 10% of events on EKS currently. Plan is to be at 100% by end of this sprint with as few 500s as possible. @hazzadous & @ellie
why? Dogfood helm chart
4. Improve onboarding experience for helm chart dev and docs @ellie
Why? Improve developer experience + will help contributions
5. Finalize and tune monitoring as code framework for helm chart @guidoiaquinti
Why? Baking in best practices for helm chart
7. Fix 40 bugs from https://github.com/PostHog/posthog/issues/11344. 4 engineers x 10 days.
Why? Make sure we have a great experience + increase retention
8. Recordings experience enhancement around quality reporting + console log. @rcmarron 
why? Make our session recording experience comprehensive/competitive
9. Complete person on events rollout @EDsCODE 
why? To unlock scale & quicker queries
10. Feature Flags Client and UX update @neilkakkar 
Why? To make our feature flag competitive

# Team sprint planning

For your team sprint planning copy this template into a comment below for each team.

```
Team ___

## Retro

<!-- Talk about what went well, what didn't go well and any actions to improve next time -->

- 

## Hang over items from previous sprint

<!-- For each item, decide to re-prioritise (and add below) or deprioritise -->

- Item 1. prioritised/deprioritise

## Planning

<!-- Each item should have a single owner. Owner can only be an engineer. -->

### High priority

-

### Bugs

-

### Low priority / side quests

-

```

**Comment by yakkomajuri** [comment]
# Team Ingestion

## Goals

1. Call the persons/groups on events done (from the ingestion side)
    - Why: To unlock high scale querying
    - Additional context: This means documentation, babysitting the fixes for the last 0.08% of events on Cloud, retesting the async migration for self-hosted
2. Unblocking large historical exports
    - Why: Big customers are requesting it
    - Additional context: Fixing timeout errors and async server flakiness, benchmarking a large export, improving its speed and reliability (in collaboration with customers)

Internal note: If we complete 1 and fix the flakiness issue on 2 it will be a successful sprint.

## Retro 

...

**Comment by fuziontech** [comment]
# Team Infra

## Retro

- @ellie joined OMG 🥳 🎆 
- @hazzadous Solid work on monitoring. Scaled events to 100% on deploy and some other situations we had a :( 500 thing. We are losing about 0.07% of events during these periods. 5 minutes tops for the intervals we are losing events. We did get to 100% 🥳  📈  Potentially have a fix with nginx. Have ideas for future improvements

## Hang over items from previous sprint

- Fixes for EKS ingestion bugs for events deployment. 500's to as close to zero as possible. @hazzadous 

## Planning

James out first week doing NC500

- We are at 10% of events on EKS currently. Plan is to be at 100% by end of this sprint with as few 500s as possible. @hazzadous & @ellie 
- Improve onboarding experience for helm chart dev and docs @ellie 
- Spend some time to spin down netlify @fuziontech 
- Finalize and tune monitoring as code framework for helm chart @guidoiaquinti 
- Come up with plan with @postgrace for rolling out Okta @fuziontech

**Comment by mariusandra** [comment]
Team App East

## Retro

Name & well or not:
- process:
- features:
- interpersonal:

Em: on a cruize with Americans in the middle of the ocean, features: though a lot, interpersonal: me and the whales.
Michael: felt productive. Focused on a solo goal, should have pulled someone else. 
Lottie: little bit lost, lot of work in docs team. New loader out with Ben. More isolated work. Going forward: 3-5 set goals/tasks so I could go and ask if needed. 
Chris: a lot of work on both teams. Process is a challenge, hiring one more product designer. Got a lot of work done. Enough rotating topics that you never get too deep on one. A lot of goals for Q3. Getting more resources for design would be good. We're working on that.
Ben: support hero for half or it. You can't do anything while it, though did some spinners with Lottie. Where we're getting to with the design system seems good. Hopefully finding the right balance between right vs in. Grateful for Micahel for his 👀. 

## Hang over items from previous sprint

- Implement "fundamental" Design System components @benjackwhite
  - A lot [implemented](https://github.com/orgs/PostHog/projects/55/views/1), some in "ready to work", most will happen over the next sprints one by one.
- Low hanging fruit for visualisations https://github.com/PostHog/posthog/issues/11106 @Twixes
  - Big bold number done
  - Some will merge into the next sprint, some deprioritised

## Planning

### High priority

Fix 40 bugs from [this list](https://github.com/PostHog/posthog/issues/11344). 4 engineers x 10 days.

### Low priority / side quests

Fix another 40 bugs.

**Comment by EDsCODE** [comment]
### Planning

**Recordings experience enhancement**
- Recording quality reporting
- Event list + console logging
- Recordings setup granularity

**Complete person on events rollout** 
- Continue and validate query correctness
- How to communicate this change best

**Feature Flags Client and UX update**
- “Immediate flag resolution for posthog-js & react-native; and UX stuff”
- Client library updates: Enable bootstrap for posthog-js and react-native
- Passing person and group properties to decide
- Decide server prop overrides & GeoIP for everyone
- UX issues
